internal/olm/operator/internal/registry_pod.go

```diff
@@ -0,0 +1,189 @@
+package olm
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"strings"
+	"time"
+
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/wait"
+	"k8s.io/client-go/kubernetes"
+)
+
+const (
+	defaultIndexImage = "quay.io/operator-framework/upstream-opm-builder:latest"
+	defaultGRPCPort   = 50051
+)
+
+// RegistryPod holds resources necessary for creation of a registry server
+type RegistryPod struct {
+	BundleAddMode string
+	BundleImage   string
+	DBPath        string
+	IndexImage    string
+	Namespace     string
+	Kubeclient    kubernetes.Interface
+
+	grpcPort int32
+}
+
+// Create returns a bundle registry pod built from an index image
+func (rp *RegistryPod) Create(ctx context.Context) (*corev1.Pod, error) {
+	// Make the pod spec
+	pod, err := rp.build()
+	if err != nil {
+		return nil, fmt.Errorf("error in building registry pod: %v", err)
+	}
+
+	// Check if Pod already exists
+	tmpPod, err := rp.Kubeclient.CoreV1().Pods(pod.Namespace).Get(ctx, pod.Name, metav1.GetOptions{})
```

Comment:
Is possible to test it locally? I mean, are you running any command to check if the Pod was created manually?

Reply:
I tested it locally using `kubectl` commands. Verified that pod creation is successful. And if there is an existing pod with the same name, instead of creating a new one, it will just return the existing one.